google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -313,34 +317,35 @@ private void handlePossibleInvalidatingException(SpannerException e) {
       res =
           executeStatementAsync(
               statement,
-              () ->
-                  runWithRetry(
-                      () -> {
-                        try {
-                          getStatementExecutor()
-                              .invokeInterceptors(
-                                  statement,
-                                  StatementExecutionStep.EXECUTE_STATEMENT,
-                                  ReadWriteTransaction.this);
-                          ResultSet delegate =
-                              DirectExecuteResultSet.ofResultSet(
-                                  internalExecuteQuery(statement, analyzeMode, options));
-                          return createAndAddRetryResultSet(
-                              delegate, statement, analyzeMode, options);
-                        } catch (AbortedException e) {
-                          throw e;
-                        } catch (SpannerException e) {
-                          createAndAddFailedQuery(e, statement, analyzeMode, options);
-                          throw e;
-                        }
-                      }),
+              () -> {
+                checkTimedOut();
```

Comment:
These changes seem bigger than they are. The only real difference is that the `checkTimeout()` call is added as the first statement of the lambda.